nick_work/fdaaa_port.py

```diff
@@ -6,367 +6,310 @@
-            else:
-                title = None
-
-            #discrep_date_status - add this to "if" statement after testing: 'and completion_date is not null')
-            if (primary_completion_date < date.today() or primary_completion_date is None) and\
-            completion_date < date.today() and\
-            study_status is ("Unknown status" or "Active, not recruiting" or "Not yet recruiting" or\
-            "Enrolling by invitation" or "Suspended" or "Recruiting"):
-                discrep_date_status = 1
-            else:
-                discrep_date_status = 0
-
-            #late_cert
-            if certificate_date is not None and certificate_date > (certificate_date + relativedelta(years=1)):
-                late_cert = 1
-            else:
-                late_cert = 0
-
-            #defaulted_date
-            if (used_primary_completion_date == 1 and defaulted_pcd_flag == 1) or\
-            used_primary_completion_date == 0 and defaulted_cd_flag == 1:
-                defaulted_date = 1
-            else:
-                defaulted_date = 0
-
-            #condition
-            try:
-                condition = json.dumps(parsed_json['clinical_study']['condition'])
-            except KeyError:
-                condition = None
-
-            #condition_mesh
-            try:
-                condition_mesh = json.dumps(parsed_json['clinical_study']['condition_browse'])
-            except KeyError:
-                condition_mesh = None
-
-            #intervention
-            try:
-                intervention = json.dumps(parsed_json['clinical_study']['intervention'])
-            except KeyError:
-                intervention = None
-
-            #intervention_mesh
-            try:
-                intervention_mesh = json.dumps(parsed_json['clinical_study']['intervention_browse'])
-            except KeyError:
-                intervention_mesh = None
-
-            #keywords
-            try:
-                keywords = json.dumps(parsed_json['clinical_study']['keyword'])
-            except KeyError:
-                keywords = None
-
-            row_list = [nct_id, act_flag, included_pact_flag, has_results, pending_results, pending_data, has_certificate,\
-                       results_due, start_date, available_completion_date, used_primary_completion_date, defaulted_pcd_flag,\
-                       defaulted_cd_flag, results_submitted_date, last_updated_date, certificate_date, phase, enrollment,\
-                       location, study_status, study_type, primary_purpose, sponsor, sponsor_type, collaborators, exported,\
-                       fda_reg_drug, fda_reg_device, is_fda_regulated, url, title, official_title, brief_title,\
-                       discrep_date_status, late_cert, defaulted_date, condition, condition_mesh, intervention, intervention_mesh,\
-                       keywords]
-
-            writer.writerow(row_list)
+    writer.writerows(zip(*trial_dict.values()))
```

Comment:
Whoa, the results of dict.values is not guaranteed to be in any order (in some versions of Python). You shouldn't disassociate the keys from values. `csv` module manages the hard part.

```python
ordered_csv_column_names = ('nct_id', 'act_flag', 'included_pact_flag', ...)

with open('test_output.csv', 'w', newline='',encoding='utf-8') as test_csv:
    csv_file = csv.DictWriter(test_raw_file, fieldnames=ordered_csv_column_names)
    csv_file.writeheader()

    for xml in ...
        s = dict()
        s["phase"] = ...

        csv.file.writerow(s)
```

Reply:
Yes, I knew this but thought for our purposes it wouldn't matter, but that's not good practice!

I've implemented your idea here all the way through and that fixes this problem!